src/domain/organizations/organizations-safe.repository.spec.ts

```diff
@@ -0,0 +1,679 @@
+      await dbWalletRepo.insert({
+        user: { id: userId },
+        address: getAddress(faker.finance.ethereumAddress()),
+      });
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await dbUserOrgRepo.insert({
+        user: { id: userId },
+        role: 'ADMIN',
+        status: 'ACTIVE',
+        organization: { id: orgId },
+      });
+
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload,
+      });
+
+      await expect(dbOrgSafeRepo.find()).resolves.toEqual(
+        payload.map(({ chainId, address }) => ({
+          id: expect.any(Number),
+          chainId,
+          address,
+          createdAt: expect.any(Date),
+          updatedAt: expect.any(Date),
+        })),
+      );
+    });
+  });
+
+  describe('findByOrganizationId', () => {
+    it('should return found organization Safes', async () => {
+      const orgSafes = faker.helpers.multiple(
+        () => ({
+          chainId: faker.string.numeric(),
+          address: getAddress(faker.finance.ethereumAddress()),
+        }),
+        { count: { min: 2, max: 5 } },
+      );
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload: orgSafes,
+      });
+
+      await expect(orgSafesRepo.findByOrganizationId(orgId)).resolves.toEqual(
+        orgSafes,
+      );
+    });
+
+    it('should return empty array if no organization Safes found', async () => {
+      await expect(
+        orgSafesRepo.findByOrganizationId(
+          faker.number.int({ max: DB_MAX_SAFE_INTEGER }),
+        ),
+      ).resolves.toEqual([]);
+    });
+  });
+
+  describe('findOrFail', () => {
+    it('should return found organizations Safes', async () => {
+      const orgSafes = faker.helpers.multiple(
+        () => ({
+          chainId: faker.string.numeric(),
+          address: getAddress(faker.finance.ethereumAddress()),
+        }),
+        { count: { min: 2, max: 5 } },
+      );
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload: orgSafes,
+      });
+
+      await expect(
+        orgSafesRepo.findOrFail({
+          where: { organization: { id: orgId } },
+        }),
+      ).resolves.toEqual(
+        expect.arrayContaining(
```

Comment:
We don't need to use `expect.arrayContaining` here.

Reply:
I've changed `orgSafesRepo.create` by `dbOrgSafeRepo.insert`, and therefore an iteration through `orgSafes` is needed. The order of the insertion depends on the promises execution so `expect.arrayContaining` the test would become flaky.